lib/questions_eval/run_mimoracle.py

```diff
@@ -0,0 +1,284 @@
+    """
+    results = {}
+    for i, j in itertools.product(
+        ["synthetic_summary", "summary"],
+        ["synthetic_question", "question"],
+    ):
+        results[f"{i}/{j}"] = evaluation_chain.invoke({"summary": row[i], "question": row[j]})
+
+    # Compute conformity, consistency and coverage
+    coverage = 1 if is_idk(results["synthetic_summary/question"]) else 0
+    consistency = 1 if not is_idk(results["summary/synthetic_question"]) else 0
+    conformity = compute_conformity(
+        results["synthetic_summary/question"],
+        results["summary/synthetic_question"],
+        results["summary/question"],
+        results["synthetic_summary/synthetic_question"],
+    )
+    results["consistency"] = consistency
+    results["conformity"] = conformity
+    results["coverage"] = coverage
+    return results
+
+
+def create_chain(template: str, llm: str, is_question_chain: bool):
+    """
+    Create a chain of models
+
+    Args:
+        template: The template for the prompt
+        llm: The language model to use
+        is_question_chain: Boolean indicating whether the chain is used for
+        question generation (True) or summary generation (False)
+
+    Returns:
+        The chain of models for either question or summary generation
+    """
+    chat_template = ChatPromptTemplate.from_messages(
+        [
+            ("system", "You are an helpful clinical assistant."),
+            ("human", template),
+        ]
+    )
+
+    return (
+        chat_template
+        | llm
+        | (
+            StrOutputParser()
+            if not is_question_chain
+            else OutputFixingParser.from_llm(llm, parser=JsonOutputParser())
+        )
+    )
+
+
+@hydra.main(config_path="./configs", config_name="run_mimoracle.yaml")
+def main(cfg: DictConfig):
+    # Initialize WandB and log the models
+    wandb.init(project="document-cross-validation", entity="clinical-dream-team")
+    wandb.config.update(OmegaConf.to_container(cfg, resolve=True))
+
+    llm = hydra.utils.instantiate(cfg.model)
+    question_llm = hydra.utils.instantiate(cfg.question_model)
+    summary_chain = create_chain(cfg.prompts.summary, llm, False)
+    question_chain = create_chain(cfg.prompts.question, question_llm, True)
+    evaluation_chain = create_chain(cfg.prompts.evaluation, question_llm, False)
+
+    # Load and process dataset
+    df = pd.read_csv(cfg.dataset).iloc[: cfg.samples]
+    # rename column
+    df.rename(columns={"section_content": "summary"}, inplace=True)
+
+    tqdm.pandas(desc="Generating data...")
+    ds_questions = [
+        item
+        for _, row in df.progress_apply(
+            generate_data,
+            axis=1,
+            args=[cfg.num_questions, summary_chain, question_chain],
+        ).items()
+        for item in row
+    ]
+    print(f"Shape of generated data: {len(ds_questions)}")
+    df_questions = pd.DataFrame(ds_questions)
```

Comment:
same as the previous ones maybe you should also separate summary processing and question processing